docs/reference/protocol-spec.md

```diff
@@ -1,214 +1,252 @@
+Redis 6.0 introduced experimental opt-in support to a subset of RESP3's features.
+In addition, the introduction of the `HELLO` command allows clients to handshake and upgrade the connection's protocol version (see [Client handshake](#client-handshake)).
 
-RESP is actually a serialization protocol that supports the following
-data types: Simple Strings, Errors, Integers, Bulk Strings, and Arrays.
+Up to and including Redis 7, both RESP2 and RESP3 clients can invoke all core commands. However, some commands return differently-typed replies for different protocol versions.
 
-Redis uses RESP as a request-response protocol in the
-following way:
+Future versions of Redis may change the default protocol version, but it is unlikely that RESP2 will become entirely deprecated.
+It is possible, however, that new features in upcoming versions will require the use of RESP3.
 
-* Clients send commands to a Redis server as a RESP Array of Bulk Strings.
-* The server replies with one of the RESP types according to the command implementation.
-
-In RESP, the first byte determines the data type:
+## Network layer
+A client connects to a Redis server by creating a TCP connection to its port (the default is 6379).
 
-* For **Simple Strings**, the first byte of the reply is "+"
-* For **Errors**, the first byte of the reply is "-"
-* For **Integers**, the first byte of the reply is ":"
-* For **Bulk Strings**, the first byte of the reply is "$"
-* For **Arrays**, the first byte of the reply is "`*`"
+While RESP is technically non-TCP specific, the protocol is used exclusively with TCP connections (or equivalent stream-oriented connections like Unix sockets) in the context of Redis.
 
-RESP can represent a Null value using a special variation of Bulk Strings or Array as specified later.
+## Request-Response model
+The Redis server accepts commands composed of different arguments.
+Then, the server processes the command and sends the reply back to the client.
+
+This is the simplest model possible; however, there are some exceptions:
+
+* Redis requests can be [pipelined](#multiple-commands-and-pipelining).
+  Pipelining enables clients to send multiple commands at once and wait for replies later.
+* When a RESP2 connection subscribes to a [Pub/Sub](/docs/manual/pubsub) channel, the protocol changes semantics and becomes a *push* protocol.
+  The client no longer requires sending commands because the server will automatically send new messages to the client (for the channels the client is subscribed to) as soon as they are received.
+* The `MONITOR` command.
+  Invoking the `MONITOR` command switches the connection to an ad-hoc push mode.
+  The protocol of this mode is not specified but is obvious to parse.
+* [Protected mode](/docs/management/security/#protected-mode).
+  Connections opened from a non-loopback address to a Redis while in protected mode are automatically sent a `-DENIED` reply and terminated by the server.
```

Comment:
Please make it clearer that the DENIED error will be emitted without issuing any command

Reply:
Better?